sqrl-execute/sqrl-execute-flink/sqrl-execute-flink-connectors/sqrl-execute-flink-file/src/main/java/com/datasqrl/io/FileSourceFactory.java

```
@@ -67,13 +72,20 @@ public SingleOutputStreamOperator<TimeAnnotatedRecord<String>> create(FlinkSourc
         builder = org.apache.flink.connector.file.src.FileSource.forRecordStreamFormat(
             format,
             FilePath.toFlinkPath(pathConfig.getDirectory()));
-        Duration monitorDuration = null;
+
         FileEnumeratorProvider fileEnumerator = new FileEnumeratorProvider(tableConfig.getBase(),
-                tableConfig.getFormat(),
-                FileDataSystemConfig.fromConfig(tableConfig));
+            tableConfig.getFormat(),
+            FileDataSystemConfig.fromConfig(tableConfig));
         builder.setFileEnumerator(fileEnumerator);
-        if (monitorDuration != null) {
-          builder.monitorContinuously(Duration.ofSeconds(10));
+
+        Optional<String> monitorInterval = tableConfig.getConnectorConfig()
+            .asString(FileSystemConnectorOptions.SOURCE_MONITOR_INTERVAL.key())
+            .getOptional();
+        if (isMonitor(monitorInterval)) {
+          Duration duration = monitorInterval
+              .map(TimeUtils::parseDuration)
+              .orElse(Duration.ofSeconds(10));
```

Should we throw an exception if an invalid duration is configured?